src/main/java/org/wso2/carbon/inbound/custom/poll/MQPollingInbound.java

```diff
@@ -0,0 +1,221 @@
+    private boolean isConnected = false;
+
+    public MQPollingInbound(Properties properties, String name, SynapseEnvironment synapseEnvironment, long scanInterval, String injectingSeq, String onErrorSeq, boolean coordination, boolean sequential) {
+
+        super(properties, name, synapseEnvironment, scanInterval, injectingSeq, onErrorSeq, coordination, sequential);
+        this.injectingSeq = injectingSeq;
+        config = new MQConfiguration(properties);
+        logger.info("Initialized the IBM-MQ inbound consumer " + name);
+    }
+
+    public Object poll() {
+
+        if (logger.isDebugEnabled()) {
+            logger.debug("Polling IBM-MQ messages for " + name);
+        }
+
+        if (!isConnected) {
+            connectionBuilder = new MQConnectionBuilder().buildConnection(config);
+            isConnected = true;
+        }
+
+        if (connectionBuilder == null) {
+            logger.error("IBM-MQ Inbound endpoint " + name + " unable to get a connection.");
+            isConnected = false;
+            return null;
+        }
+
+        manager = connectionBuilder.getManager();
+
+        MQQueue readableQueue;
+
+        if (!manager.isConnected()) {
+            logger.error("IBM-MQ Inbound endpoint queue manager not connected");
+            return null;
+        }
+        try {
+            readableQueue = manager.accessQueue(config.getQueue(), CMQC.MQRC_READ_AHEAD_MSGS);
+
+            if (readableQueue == null) {
+                logger.info("Queue is not initialized properly");
+            } else {
+
+                MQMessage message = new MQMessage();
+                MQGetMessageOptions gmo = new MQGetMessageOptions();
+
+                if (config.getCorrelationID() != null && config.getMessageID() != null) {
+                    gmo.matchOptions = MQConstants.MQMO_MATCH_CORREL_ID + MQConstants.MQMO_MATCH_GROUP_ID;
+                } else if (config.getMessageID() != null) {
+                    gmo.matchOptions = MQConstants.MQMO_MATCH_MSG_ID;
+                } else if (config.getCorrelationID() != null) {
+                    gmo.matchOptions = MQConstants.MQMO_MATCH_CORREL_ID;
+                }
+
+                readableQueue.get(message, gmo);
+                logger.info("A message received");
+
+                MQMD md = new MQMD();
+                md.copyFrom(message);
+
+                message.getDataLength();
+                int strLen = message.getDataLength();
+                byte[] strData = new byte[strLen];
+                message.readFully(strData);
+
+                //crating new message context
+                msgCtx = this.createMessageContext();
+                msgCtx.setProperty("MessageId", new String(message.messageId));
+                msgCtx.setProperty("CorrelationID", new String(message.correlationId));
+                logger.info("Messsage data-" + new String(strData));
+
+                if (message.getStringProperty("ContentType") != null) {
```

Comment:
Please check whether the correct message builders are invoked for IBM-MQ content types. Please test this at least for json and xml content types.